### PR TITLE
Portal e2e doc unit actions 

### PR DESCRIPTION
**Pull request by qijianshuai**
1. Cloud Native Security Inspector Front-End Developer Testing Guide
2. Add GitHub actions for portal unit tests
3. Added cypress badge

### CODE REVIEW

File: src/frontend/test/ui_unit.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -x
+set -e
+
+npm install -g -q --no-progress @angular/cli
+npm install -g -q --no-progress karma
+npm install -q --no-progress
+# check code lint first then run ut test
+# npm run lint
+# npm run lint:style
```
Comment (JingChen23) [inline src/frontend/test/ui_unit.sh:10]: Why we do not run lint?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -47,3 +47,65 @@ jobs:
...
+  #         npm run test
+
+  #     - uses: codecov/codecov-action@v3
+  #       with:
+  #         token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
+  #         files: ${{ github.repository }}/src/frontend/coverage/icov.info
+  #         flags: unittests # optional
+  #         name: codecov-umbrella # optional
+  #         fail_ci_if_error: true # optional (default = false)
+  #         verbose: true # optional (default = false)
```
[the remark this answers is not in the document]
Comment (qijianshuai) [reply .github/workflows/main.yml]: i will delete it

File: src/frontend/karma.conf.js [hunk 3]
```diff
@@ -1,11 +1,3 @@
-// Copyright 2022 VMware, Inc.
-// SPDX-License-Identifier: Apache-2.0
-// Karma configuration file, see link for more information
-// https://karma-runner.github.io/1.0/config/configuration-file.html
-
-// Karma configuration file, see link for more information
-// https://karma-runner.github.io/1.0/config/configuration-file.html
-
```
[the remark this answers is not in the document]
Comment (qijianshuai) [reply src/frontend/karma.conf.js]: restored

File: src/frontend/test/ui_unit.sh (hunk 1 above)
Comment (qijianshuai) [reply src/frontend/test/ui_unit.sh]: has increased

**Review by JingChen23 — APPROVED** [review]

**Pull request merged**